packages/react/src/utils/create-ts-config.ts

```diff
@@ -0,0 +1,111 @@
+import { Tree } from 'nx/src/generators/tree';
+import { readJson, updateJson } from 'nx/src/generators/utils/json';
+
+const tsConfigBaseOptions = {
+  rootDir: '.',
+  sourceMap: true,
+  declaration: false,
+  moduleResolution: 'node',
+  emitDecoratorMetadata: true,
+  experimentalDecorators: true,
+  importHelpers: true,
+  target: 'es2015',
+  module: 'esnext',
+  lib: ['es2017', 'dom'],
+  skipLibCheck: true,
+  skipDefaultLibCheck: true,
+  baseUrl: '.',
+};
+
+export function createTsConfig(
+  host: Tree,
+  projectRoot: string,
+  type: 'app' | 'lib',
+  options: {
+    strict?: boolean;
+    style?: string;
+    bundler?: string;
+    rootProject?: boolean;
+  },
+  relativePathToRootTsConfig: string
+) {
+  const json = {
+    compilerOptions: {
+      jsx: 'react-jsx',
+      allowJs: false,
+      esModuleInterop: false,
+      allowSyntheticDefaultImports: true,
+      strict: options.strict,
+    },
+    files: [],
+    include: [],
+    references: [
+      {
+        path: type === 'app' ? './tsconfig.app.json' : './tsconfig.lib.json',
+      },
+    ],
+  } as any;
+
+  if (options.style === '@emotion/styled') {
+    json.compilerOptions.jsxImportSource = '@emotion/react';
+  }
+
+  if (options.bundler === 'vite') {
+    json.compilerOptions.types = ['vite/client'];
+  }
+
+  // inline tsconfig.base.json into the project
+  if (options.rootProject) {
+    json.compileOnSave = false;
+    json.compilerOptions = { ...tsConfigBaseOptions, ...json.compilerOptions };
+    json.exclude = ['node_modules', 'tmp'];
+  } else {
+    json.extends = relativePathToRootTsConfig;
+  }
+
+  host.write(`${projectRoot}/tsconfig.json`, JSON.stringify(json, null, 2));
+}
+
+export function extractTsConfigBase(host: Tree) {
+  if (host.exists('tsconfig.base.json')) return;
+
+  const tsconfig = readJson(host, 'tsconfig.json');
+  const baseCompilerOptions = {} as any;
+  for (let compilerOption of Object.keys(tsConfigBaseOptions)) {
```

Comment:
I feel like we should have this logic in 1 place, not the responsibility of `angular` or `react`. It's the responsibility of `@nrwl/workspace`.

For `jest`, `@nrwl/jest` owns the logic.  For `linter`, `@nrwl/linter` owns it.

When it comes time to do `nest`, we shouldn't have to duplicate this logic yet again.

Reply:
I moved it